MapWindow.cpp

```diff
@@ -590,13 +590,29 @@ const char *MapWindow::look(uint16 x, uint16 y, bool show_prefix)
 }
 
 
-Obj *MapWindow::get_objAtCursor()
+Obj *MapWindow::get_objAtCursor(bool for_use /* = false */)
 {
+    uint16 tile_n = tmp_map_buf[(cursor_y+TMP_MAP_BORDER) * tmp_map_width + (cursor_x+TMP_MAP_BORDER)];
+    if(tile_n == 0) //black area
+        return NULL; // nothing to see here. ;)
 
- if(tmp_map_buf[(cursor_y+TMP_MAP_BORDER) * tmp_map_width + (cursor_x+TMP_MAP_BORDER)] == 0) //black area
-   return NULL; // nothing to see here. ;)
-
- return obj_manager->get_obj(WRAPPED_COORD(cur_x + cursor_x, cur_level), cur_y + cursor_y, cur_level,OBJ_SEARCH_TOP, OBJ_EXCLUDE_IGNORED);
+    Obj *obj = obj_manager->get_obj(WRAPPED_COORD(cur_x + cursor_x, cur_level), cur_y + cursor_y, cur_level,OBJ_SEARCH_TOP, OBJ_EXCLUDE_IGNORED);
+    // Savage Empire Create Object from Tile
+    if (for_use && game_type == NUVIE_GAME_SE && obj == NULL)
+    {
+        Script *script = game->get_script();
+        Uint16 obj_n = script->call_get_tile_to_object_mapping(tile_n);
+        printf("mapped obj #: %x\n", obj_n);
+        if (obj_n != 0) {
+            obj = new Obj();
```

Comment:
Will this Obj leak? It doesn't look like it is managed by the game engine.

Reply:
Yes.  I wrote this a couple years ago and misremembered what I did.  Getting back into this and wanted to get his first set of changes into the code base since there were several changes.  I had thought I'd implemented a non-leak version that kept a single object for each tile type, but I didn't do that.  Sorry.  Will fix and resubmit soon

Reply:
The easier way to resolve this for me (not understanding the object manager at all) is to keep a static object for each tile type.  If that is acceptable for now, I will implement that.

Reply:
I implemented the fix as well as fixed a minor bug with changes to Event.cpp since I originally wrote this code.